Finished tasks 7-3, 7-4, 8-1.

prychkov opened the pull request.

prychkov committed:
Merge branch 'master' of https://github.com/js-tasks-ru/jsbasic-20210906-2_rychkovpavel

prychkov committed:
Merge branch 'master' of https://github.com/js-tasks-ru/jsbasic-20210906-2_rychkovpavel

prychkov committed:
Merge branch 'master' of https://github.com/js-tasks-ru/jsbasic-20210906-2_rychkovpavel

prychkov committed:
completed 7 module 3 taska

prychkov committed:
finished tasks 7-3, 7-4, 8-1

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

SinimaWath commented on 7-module/4-task/index.js:
```diff
@@ -1,4 +1,320 @@
   }
 }
+
+// Второй вариант
```
Второй вариант по лучше ;)

SinimaWath reviewed (changes requested):
тесты не проходят ( На внешний вид все ок

prychkov committed:
8-2 filter

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

prychkov committed:
filter products 8-2

prychkov committed:
task 8-3 completed

SinimaWath commented on 8-module/2-task/index.js:
```diff
@@ -5,5 +5,52 @@ export default class ProductGrid {
+    let productsGridInner = document.querySelector('.products-grid__inner');
+    productsGridInner.innerHTML = "";
+
+    Object.assign(this.filters, filters);
```
Тут правильно будет вот так
`this.filters = Object.assign(this.filters, filters);`

Object.assign - возвращает результат и не изменяет  аргументы

prychkov committed:
task 8.4.

prychkov committed:
task 8.4. completed

prychkov committed:
change function updateProductCount

prychkov committed:
9.4 task complited

prychkov committed:
change tasks: 7.3 and 7.4, in function setBasicSettings that set thumb and progress, earlier do not do right value

prychkov committed:
new token_github

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

prychkov committed:
deleted removed unnecessary imports createElement and ProductCard

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

prychkov committed:
task 8.2.
replaced 'let productsGridInner = document.querySelector('.products-grid__inner');'
with 'let productsGridInner = this._container.querySelector('.products-grid__inner');'
becouse unit tests returned error - TypeError: Cannot set properties of null (setting 'innerHTML')

jsru commented:
Добавляю преподавателя (@SinimaWath) для код-ревью.
